src/SourceBuild/content/eng/tools/join-verticals.proj

```diff
@@ -0,0 +1,46 @@
+<Project>
+    <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
+
+    <PropertyGroup>
+        <!-- Fake, to satisfy the SDK. -->
+        <TargetFramework>netstandard2.0</TargetFramework>
+        <DisableImplicitFrameworkReferences>true</DisableImplicitFrameworkReferences>
+    </PropertyGroup>
+
+    <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
+
+    <Target Name="Build">
+        <MSBuild Projects="$(ToolsDir)init-build.proj"
+             Targets="Build"
+             StopOnFirstFailure="true" />
+    </Target>
+
+    <UsingTask TaskName="Microsoft.DotNet.UnifiedBuild.Tasks.JoinVerticals" AssemblyFile="$(MicrosoftDotNetUnifiedBuildTasksAssembly)" />
+    <Target Name="JoinVerticals " AfterTargets="Build">
```

Comment:
I would call this project "publish-final.proj" and the target "PublishFinal":

```suggestion
    <Target Name="PublishFinal" AfterTargets="Build">
```

Reply:
I don't think we should call it `publish-final`, as this will also be used to download artifacts for join point builds like we discussed. I named it `join-verticals` since it can join any number of verticals, including all of them

Reply:
I was under the impression that we can't directly use this target for the join but if we can, great.

Reply:
We should be able to, there's a property called `VerticalSubSet` that has a list of verticals we want to join, so it doesn't have to be the full set. On second thought, this isn't a very good name, perhaps `VerticalsToJoin` is better. 
There's also a `FlatCopy` property, that tells the task how we want it to copy artifacts, if it's set to true, all packages will be in a flat layout in the packages folder, and if it's not, we'll keep the same layout as it is in the artifacts, so for the non final join points we should keep it false. Same goes for assets